pandas/core/window.py

```diff
@@ -1913,19 +1919,13 @@ def _get_window(self, other=None):
     7  0.680292  0.132049  0.548693
     8  0.067236  0.948257  0.163353
     9 -0.286980  0.618493 -0.694496
-
-    See Also
-    --------
-    pandas.DataFrame.expanding.aggregate
-    pandas.DataFrame.rolling.aggregate
-    pandas.DataFrame.aggregate
     """)
 
-    @Appender(_agg_doc)
     @Appender(_shared_docs['aggregate'] % dict(
         versionadded='',
         klass='Series/DataFrame',
         axis=''))
+    @Appender(_agg_doc)
     def aggregate(self, arg, *args, **kwargs):
```

Comment:
do you mind generating the docstrings of this with `./scripts/validate_docstrings.py pandas.core.window.Expanding.aggregate` to see the final result of this? I guess you did, but if you can copy the output in a comment, so I can understand better this change, that would be great.

Reply:
I think these changes are a mistake on my behalf. 

I want to retain the order of _Parameters_, _Returns_, _See Also_, _Notes_, _Examples_, but _shared_docs['aggregate'] already has a _notes_ section directly following _Parameters_ and _Returns_. 

This means I need to separate __agg_doc_ into two strings for _See Also_ and _Examples_, which can be inserted in the correct order with _shared_docs['aggregate']. i.e. the order and where the string comes from, would be as follows:

_Parameters_ (_shared_docs['aggregate']), _Returns_ (_shared_docs['aggregate']), _See Also_ (__agg_doc),  _Notes_ (_shared_docs['aggregate']), _Examples_ (__agg_doc).

I looked at creating a %(see_also)s in _shared_docs['aggregate'] to substitute in a separated _See Also_, but this leads to KeyErrors for other docstrings using _shared_docs['aggregate']. 

I am likely over complicating this, open to your suggestions!

Reply:
Here's the output of ./scripts/validate_docstrings.py pandas.core.window.Expanding.aggregate

As you can see, the order is almost correct but for any method docstring that uses @Appender(_shared_docs['aggregate']) - the Notes section will always appear before See Also - which is the incorrect order - this is the issue I was attempting to describe above.

`################################################################################
############## Docstring (pandas.core.window.Expanding.aggregate) ##############
################################################################################

Aggregate using one or more operations over the specified axis.



Parameters
----------
func : function, str, list or dict
    Function to use for aggregating the data. If a function, must either
    work when passed a Series/DataFrame or when passed to Series/DataFrame.apply.

    Accepted combinations are:

    - function
    - string function name
    - list of functions and/or function names, e.g. ``[np.sum, 'mean']``
    - dict of axis labels -> functions, function names or list of such.

*args
    Positional arguments to pass to `func`.
**kwargs
    Keyword arguments to pass to `func`.

Returns
-------
DataFrame, Series or scalar
    if DataFrame.agg is called with a single function, returns a Series
    if DataFrame.agg is called with several functions, returns a DataFrame
    if Series.agg is called with single function, returns a scalar
    if Series.agg is called with several functions, returns a Series

Notes
-----
`agg` is an alias for `aggregate`. Use the alias.

A passed user-defined-function will be passed a Series for evaluation.

See Also
--------
pandas.DataFrame.expanding.aggregate
pandas.DataFrame.rolling.aggregate
pandas.DataFrame.aggregate

Examples
--------

>>> df = pd.DataFrame(np.random.randn(10, 3), columns=['A', 'B', 'C'])
>>> df
          A         B         C
0 -2.385977 -0.102758  0.438822
1 -1.004295  0.905829 -0.954544
2  0.735167 -0.165272 -1.619346
3 -0.702657 -1.340923 -0.706334
4 -0.246845  0.211596 -0.901819
5  2.463718  3.157577 -1.380906
6 -1.142255  2.340594 -0.039875
7  1.396598 -1.647453  1.677227
8 -0.543425  1.761277 -0.220481
9 -0.640505  0.289374 -1.550670

>>> df.ewm(alpha=0.5).mean()
          A         B         C
0 -2.385977 -0.102758  0.438822
1 -1.464856  0.569633 -0.490089
2 -0.207700  0.149687 -1.135379
3 -0.471677 -0.645305 -0.906555
4 -0.355635 -0.203033 -0.904111
5  1.076417  1.503943 -1.146293
6 -0.041654  1.925562 -0.588728
7  0.680292  0.132049  0.548693
8  0.067236  0.948257  0.163353
9 -0.286980  0.618493 -0.694496

################################################################################
################################## Validation ##################################
################################################################################

11 Errors found:
	Double line break found; please use only one blank line to separate sections or paragraphs, and do not leave blank lines at the end of docstrings
	Sections are in the wrong order. Correct order is: Parameters, Returns, See Also, Notes, Examples
	Parameters {arg} not documented
	Unknown parameters {func}
	Missing description for See Also "pandas.DataFrame.expanding.aggregate" reference
	pandas.DataFrame.expanding.aggregate in `See Also` section does not need `pandas` prefix, use DataFrame.expanding.aggregate instead.
	Missing description for See Also "pandas.DataFrame.rolling.aggregate" reference
	pandas.DataFrame.rolling.aggregate in `See Also` section does not need `pandas` prefix, use DataFrame.rolling.aggregate instead.
	Missing description for See Also "pandas.DataFrame.aggregate" reference
	pandas.DataFrame.aggregate in `See Also` section does not need `pandas` prefix, use DataFrame.aggregate instead.
	Examples do not pass tests
1 Warnings found:
	No extended summary found

################################################################################
################################### Doctests ###################################
################################################################################

**********************************************************************
Line 48, in pandas.core.window.Expanding.aggregate
Failed example:
    df
Expected:
              A         B         C
    0 -2.385977 -0.102758  0.438822
    1 -1.004295  0.905829 -0.954544
    2  0.735167 -0.165272 -1.619346
    3 -0.702657 -1.340923 -0.706334
    4 -0.246845  0.211596 -0.901819
    5  2.463718  3.157577 -1.380906
    6 -1.142255  2.340594 -0.039875
    7  1.396598 -1.647453  1.677227
    8 -0.543425  1.761277 -0.220481
    9 -0.640505  0.289374 -1.550670
Got:
              A         B         C
    0 -0.248355 -0.790248  1.093674
    1  1.639693 -0.920766  0.438297
    2 -0.201612  0.528843 -0.659469
    3 -1.613252 -0.498679  2.970533
    4 -1.210187  0.109458  0.226834
    5 -1.055925  0.232102 -0.212216
    6  1.119070 -0.183378  0.222235
    7 -0.043799 -0.486759 -1.448347
    8  1.288434  0.889187  0.570584
    9  0.346035  0.229559  0.595709
**********************************************************************
Line 61, in pandas.core.window.Expanding.aggregate
Failed example:
    df.ewm(alpha=0.5).mean()
Expected:
              A         B         C
    0 -2.385977 -0.102758  0.438822
    1 -1.464856  0.569633 -0.490089
    2 -0.207700  0.149687 -1.135379
    3 -0.471677 -0.645305 -0.906555
    4 -0.355635 -0.203033 -0.904111
    5  1.076417  1.503943 -1.146293
    6 -0.041654  1.925562 -0.588728
    7  0.680292  0.132049  0.548693
    8  0.067236  0.948257  0.163353
    9 -0.286980  0.618493 -0.694496
Got:
              A         B         C
    0 -0.248355 -0.790248  1.093674
    1  1.010343 -0.877260  0.656756
    2  0.317798 -0.073773 -0.095373
    3 -0.712096 -0.300389  1.539777
    4 -0.969175 -0.088855  0.862129
    5 -1.013238  0.074171  0.316430
    6  0.061311 -0.055618  0.268962
    7  0.008550 -0.272034 -0.593060
    8  0.649744  0.309713 -0.010099
    9  0.497741  0.269597  0.293101`